h2o-core/src/main/java/water/util/Log.java

```diff
@@ -22,277 +27,205 @@
-    return logFileName;
+    return "h2o_" + ip + "_" + portString;
   }
 
-  /**
-   * @return This is what shows up in the Web UI when clicking on show log file.  File name only.
-   */
-  public static String getLogFileName(String level) throws Exception {
-    String f;
-    switch (level) {
-      case "trace": f = "-1-trace.log"; break;
-      case "debug": f = "-2-debug.log"; break;
-      case "info":  f = "-3-info.log"; break;
-      case "warn":  f = "-4-warn.log"; break;
-      case "error": f = "-5-error.log"; break;
-      case "fatal": f = "-6-fatal.log"; break;
-      case "httpd": f = "-httpd.log"; break;
-      default:
-        throw new Exception("Unknown level");
-    }
+  private static void setPropertiesForLevel(Properties p, String logger, Level level) {
+   setPropertiesFor(p, logger, "%m%n", level.name(), getLogFilePath(level));
+  }
 
-    return getLogFileNameStem() + f;
+  private static void setPropertiesForHTTPD(Properties p){
+    p.setProperty("log4j.logger.water.api.RequestServer",       "TRACE, " + HTTPD);
+    p.setProperty("log4j.additivity.water.api.RequestServer",   "false");
+    setPropertiesFor(p, HTTPD, "%d{ISO8601} %m%n", HTTPD.toLowerCase(), getLogFilePathHTTPD());
   }
 
-  private static void setLog4jProperties(String logDir, java.util.Properties p) throws Exception {
-    LOG_DIR = logDir;
-    String logPathFileName = getLogPathFileNameStem();
+  private static void setPropertiesFor(Properties p, String logger, String pattern, String level, String logPath){
+    p.setProperty("log4j.appender." + logger,                               "org.apache.log4j.RollingFileAppender");
+    p.setProperty("log4j.appender." + logger + ".Threshold",                level);
+    p.setProperty("log4j.appender." + logger + ".File",                     logPath);
+    p.setProperty("log4j.appender." + logger + ".MaxFileSize",              "1MB");
+    p.setProperty("log4j.appender." + logger + ".MaxBackupIndex",           "3");
+    p.setProperty("log4j.appender." + logger + ".layout",                   "org.apache.log4j.PatternLayout");
+    p.setProperty("log4j.appender." + logger + ".layout.ConversionPattern",  pattern);
+  }
 
-    // H2O-wide logging
+  private static void setLog4jProperties(Properties p) {
     String appenders = new String[]{
-      "TRACE, R6",
-      "TRACE, R5, R6",
-      "TRACE, R4, R5, R6",
-      "TRACE, R3, R4, R5, R6",
-      "TRACE, R2, R3, R4, R5, R6",
-      "TRACE, R1, R2, R3, R4, R5, R6",
-    }[_level];
+            "TRACE, R6",
+            "TRACE, R5, R6",
+            "TRACE, R4, R5, R6",
+            "TRACE, R3, R4, R5, R6",
+            "TRACE, R2, R3, R4, R5, R6",
+            "TRACE, R1, R2, R3, R4, R5, R6",
+    }[currentLevel.numLevel];
     p.setProperty("log4j.logger.water.default", appenders);
     p.setProperty("log4j.additivity.water.default",   "false");
 
-    p.setProperty("log4j.appender.R1",                          "org.apache.log4j.RollingFileAppender");
```

Comment:
I implemented it based on @mmalohlava suggestion. The code was kind of repetitive so wanted to get rid of that and have one single place where we configure it.

I can however change it back, I have no strong opinion on either original or new approach. The original is easier to read, to second one is less code with single point of configuration.